melodic-manatees/early_internet/multitab/views.py

```diff
@@ -0,0 +1,98 @@
+from django.shortcuts import render, redirect
+from .models import MultiTab
+from django.forms import modelformset_factory
+from django.contrib.auth.decorators import login_required
+'''Views for the MultiTab app'''
+
+
+@login_required(login_url='login')
+def multitab_home(request):
+    # get all MultiTab objects for the current user
+    multitabs = MultiTab.objects.filter(multitab_owner=request.user.profile)
+    if multitabs:
+        if len(multitabs) == 1:
+            title_1 = multitabs[0].title
+            site_1_1 = multitabs[0].tab_one
+            site_1_2 = multitabs[0].tab_two
+            site_1_3 = multitabs[0].tab_three
+            title_2 = 'Multitab'
+            site_2_1 = ''
+            site_2_2 = ''
+            site_2_3 = ''
+        elif len(multitabs) == 2:
+            title_1 = multitabs[0].title
+            site_1_1 = multitabs[0].tab_one
+            site_1_2 = multitabs[0].tab_two
+            site_1_3 = multitabs[0].tab_three
+            title_2 = multitabs[1].title
+            site_2_1 = multitabs[1].tab_one
+            site_2_2 = multitabs[1].tab_two
+            site_2_3 = multitabs[1].tab_three
+    else:
+        title_1 = 'Multitab'
+        site_1_1 = ''
+        site_1_2 = ''
+        site_1_3 = ''
+        title_2 = 'Multitab'
+        site_2_1 = ''
+        site_2_2 = ''
+        site_2_3 = ''
```

Comment:
This code is a mess. These variable names are quite ugly, and you should've just been constructing a dictionary here and passing it into the `render` later instead of making 9 variables to pass into the dict later.

Here's one way you could've done this cleaner:
```py
context = {
    'title_1': 'Multitab',
    'site_1_1': '',
    'site_1_2': '',
    'site_1_3': '',
    'title_2': 'Multitab',
    'site_2_1': '',
    'site_2_2': '',
    'site_2_3': '',
}

for num, multitab in enumerate(multitabs):
    context[f'title_{num + 1}'] = multitab[num].title
    context[f'site_{num + 1}_1'] = multitab[num].tab_one
    context[f'site_{num + 1}_2'] = multitab[num].tab_two
    context[f'site_{num + 1}_3'] = multitab[num].tab_three
```

This has less nesting, and also has the added benefit of working for any number of multitabs.